crates/flowistry/src/source_map.rs

```diff
@@ -482,27 +501,18 @@ mod test {
     check("actual", missing_actual);
   }
 
-  #[test]
-  fn test_hir_spanner() {
-    let src = r#"fn foo() {
-      let x = `(1)`;
-      let `(y)` = x + 1;
-      if `(true)``( )`{ let z = 1; }
-      x `(+)` y;
-    }"#;
+  fn hir_spanner_harness(
+    src: &str,
+    expected: &[&[&str]],
+    enclosing_spans: EnclosingHirSpans,
+  ) {
     harness(src, |tcx, body_id, body, spans| {
       let spanner = HirSpanner::new(tcx, body_id);
       let source_map = tcx.sess.source_map();
-      let expected: &[&[&str]] = &[
-        &["1"],
-        &["let y = ", ";"],
-        &["true"],
-        &["if ", " { ", " }"],
-        &[" + "],
-      ];
       debug!("{}", body.to_string(tcx).unwrap());
       for (input_span, desired) in spans.into_iter().zip(expected) {
-        let mut enclosing = spanner.find_enclosing_hir(input_span);
+        let mut enclosing =
+          spanner.find_enclosing_hir(input_span, enclosing_spans.clone());
```

Comment:
I would add `Copy` to the derive on `EnclosingHirSpans`, and then you don't need to `.clone()` here.